### PR TITLE
MAT-9904: Utilize VSES in Validations

### PR DESCRIPTION
**Pull request by chubert-sb**
## MADiE FHIR SERVICE

Jira Ticket: [MAT-9904](https://jira.cms.gov/browse/MAT-9904)
(Optional) Related Tickets:

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/madie/madiefhirservice/config/HapiFhirConfig.java [hunk 1]
```diff
@@ -94,21 +100,24 @@ public IValidationSupport validationSupportChainQiCore600(
...
+        new VSESValidationSupport(
+            qicore6FhirContext,
+            savedExpansionServiceBaseUrl,
+            new BasicAuthInterceptor("api-key", savedExpansionServiceApiKey),
+            validationConfig);
+
     return new ValidationSupportChain(
-        prePopulatedValidationSupport,
         npmPackageSupport,
+        vsesValidationSupport,
```
Comment (RohitKandimalla) [inline src/main/java/gov/cms/madie/madiefhirservice/config/HapiFhirConfig.java:119]: Should we do it after DefaultProfileValidationSupport?

**Review by RohitKandimalla — DISMISSED** [review]

File: src/main/java/gov/cms/madie/madiefhirservice/config/HapiFhirConfig.java (hunk 1 above)
Comment (chubert-sb) [reply src/main/java/gov/cms/madie/madiefhirservice/config/HapiFhirConfig.java:119]: I originally had it after DefaultProfileValidationSupport, but it will return null results and validation will do some conversion and accept it as a real validation instead of passing to the next chain. So, I moved it here to fix that.

File: src/main/java/gov/cms/madie/madiefhirservice/validators/VSESValidationSupport.java [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+import org.hl7.fhir.r4.model.ValueSet;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Validation support module that fetches previously-saved ValueSet expansions from an external
+ * service. The service is expected to expose an endpoint that returns a serialized ValueSet
+ * (preferably FHIR JSON) for a given ValueSet URL. service. The service is expected to expose an
+ * endpoint that returns a serialized ValueSet (preferably FHIR JSON) for a given ValueSet URL.
```
Comment (jkotanchik-SB) [inline src/main/java/gov/cms/madie/madiefhirservice/validators/VSESValidationSupport.java]:
```suggestion
 * (preferably FHIR JSON) for a given ValueSet URL.
```
Duplicated text.

**Review by jkotanchik-SB — DISMISSED** [review]

**Review by jkotanchik-SB — APPROVED** [review]
